lib/client-assets.php

```diff
@@ -1093,9 +1093,25 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
  * Remove this in Gutenberg 3.1
  */
 function polyfill_blocks_module_in_scripts() {
-	if ( is_admin() ) {
-		wp_enqueue_script( 'wp-editor' );
+	global $post;
+
+	if ( ! is_admin() ) {
+		return;
 	}
+
+	if ( get_current_screen()->base !== 'post' ) {
+		return;
+	}
+
+	if ( isset( $_GET['classic-editor'] ) ) {
+		return;
+	}
+
+	if ( ! gutenberg_can_edit_post( $post ) ) {
+		return;
+	}
```

Comment:
It might be handy to define a `is_editor_page` function to check whether we're loading a Gutenberg page or not